app/package/controllers.py

```diff
@@ -185,7 +206,63 @@ def delete_data_package(publisher, package):
     try:
         bitstore = BitStore(publisher=publisher, package=package)
         status_acl = bitstore.change_acl('private')
-        status_db = MetaDataDB.change_status(publisher, package)
+        status_db = Package.change_status(publisher, package)
+        if status_acl and status_db:
+            return jsonify({"status": "OK"}), 200
+        if not status_acl:
+            raise Exception('Failed to change acl')
+        if not status_db:
+            raise Exception('Failed to change status')
+    except Exception as e:
+        app.logger.error(e)
+        return handle_error('GENERIC_ERROR', e.message, 500)
+
+
+@package_blueprint.route("/<publisher>/<package>/undelete", methods=["POST"])
+@requires_auth
+@is_allowed('Package::Undelete')
+def undelete_data_package(publisher, package):
+    """
+    DPR data package un-delete operation.
+    This API is responsible for un-mark the mark for delete of data package
+    ---
+    tags:
+        - package
+    parameters:
+        - in: path
+          name: publisher
+          type: string
+          required: true
+          description: publisher name
+        - in: path
+          name: package
+          type: string
+          required: true
+          description: package name
+        - in: header
+          name: Authorization
+          type: string
+          required: true
+          description: >
+            Jwt token in format of "bearer {token}.
+            The token can be generated from /api/auth/token"
+    responses:
+        500:
+            description: Internal Server Error
+        200:
+            description: Success Message
+            schema:
+                id: put_package_success
+                properties:
+                    status:
+                        type: string
+                        default: OK
+
+    """
+    try:
+        bitstore = BitStore(publisher=publisher, package=package)
+        status_acl = bitstore.change_acl('public-read')
+        status_db = Package.change_status(publisher, package, status='active')
```

Comment:
@Fak3 ```db.Enum``` would be good if converted to boolean we would not add any more state. Thanks